std/array.d

```diff
@@ -342,6 +345,105 @@ then the result will contain the value of the last pair for that key in r.
 See_Also: $(XREF typecons, Tuple)
  */
 
+/**
+ * Initialize a static array of the given size from the range $(D r).
+ *
+ * If `r` has more than `n` elements, only the first `n` are copied.
+ * If `r` has less than `n` elements, the remaining spaces in the static array
+ * will have default values.
+ *
+ * Params:
+ *      r = range (or aggregate with $(D opApply) function) whose elements are
+ *      copied into the static array.
+ *
+ * Returns:
+ *      static array of size `n`
+ */
+ForeachType!Range[n] staticArray(size_t n, Range)(Range r)
+if (isIterable!Range)
+{
+    typeof(return) result;
+
+    // manually count our index instead of using foreach(i, e; r) because:
+    // 1. for a range, we would need to use enumerate
+    // 2. for opApply with no indexed overload, we would need to count anyways
+    size_t i;
+    foreach (e; r)
+    {
+        if (i >= n) break; // array is full, ignore remaining elements
+        result[i++] = e;
+    }
+
+    return result;
+}
+
+///
+@safe pure nothrow unittest
+{
+    import std.range : only;
+    int[3] a = only(1, 2, 3).staticArray!3;
+    assert(a == [ 1, 2, 3 ]);
+}
+
+/// `staticArray!n` only takes `n` elements, even if the range has (infinitely) more.
+@safe pure nothrow unittest
+{
+    import std.range : repeat;
+    int[4] a = repeat(5).staticArray!4;
+    assert(a == [ 5, 5, 5, 5 ]);
+}
+
+/// `staticArray!n` leaves remaining elements default if the range is smaller than `n`
+@safe pure nothrow unittest
+{
+    import std.range : only;
+    int[5] a = only(1,2,3).staticArray!5;
+    assert(a == [ 1, 2, 3, 0, 0 ]);
+}
+
+/// `staticArray!n` works with `opApply` as well as a range interface
+unittest
+{
+    static struct OpApply
+    {
+        int opApply(int delegate(ref int) dg)
+        {
+            int res;
+            foreach(i; 0..10)
+            {
+                res = dg(i);
+                if(res) break;
+            }
+
+            return res;
+        }
+    }
+
+    int[5] a = OpApply().staticArray!5;
+    assert(a == [ 0, 1, 2, 3, 4 ]);
+}
+
+/**
+ * Convert an array literal to a static array.
+ *
+ * Params:
+ *      arr = an array literal
+ *
+ * Returns: static array of size `arr.length`
+ */
+T[n] staticArray(T, size_t n)(T[n] arr)
+{
+	return arr;
+}
+
+///
+@safe pure nothrow unittest
+{
+    auto i = [1,2,3,4].staticArray;         // no need to declare size
+    static assert(is(typeof(i) == int[4])); // i is a static array
+    assert(i == [1,2,3,4]);
```

Comment:
Should rename `i` to `arr` or `sa`. Also `a` in the other unit tests should be `arr` or `sa`.